test/test_common/threadsafe_singleton_injector.h

```diff
@@ -17,4 +17,24 @@ template <class T> class TestThreadsafeSingletonInjector {
   T* latched_instance_;
 };
 
+// Note this class is not thread-safe, and should be called exceedingly carefully.
+template <class T> class TestScopedInjectableLoader {
+public:
+  TestScopedInjectableLoader(std::unique_ptr<T>&& instance)
```

Comment:
This looks a lot like ScopedInjectableLoader
Do we actually need to restore the prior singleton and if so do you think we should just enhance or inherit from that class?

Reply:
The swapping is needed in unit tests, where one test needs to swap out the singleton but other do not.
I had another PR where I have modified the `ScopedInjectableLoader`, but since this behavior is limited to test only a test specific class is I think better.